stacks-signer/src/signer.rs

```diff
@@ -0,0 +1,2005 @@
+                    "Signer #{}: Received a nonce timeout error.",
+                    self.signer_id
+                );
+            }
+            SignError::InsufficientSigners(malicious_signers) => {
+                debug!(
+                    "Signer #{}: Received a insufficient signers error.",
+                    self.signer_id
+                );
+                let message = self.coordinator.get_message();
+                let block = read_next::<NakamotoBlock, _>(&mut &message[..]).ok().unwrap_or({
+                        // This is not a block so maybe its across its hash
+                        // This jankiness is because a coordinator could have signed a rejection we need to find the underlying block hash
+                        let signer_signature_hash_bytes = if message.len() > 32 {
+                            &message[..32]
+                        } else {
+                            &message
+                        };
+                        let Some(signer_signature_hash) = Sha512Trunc256Sum::from_bytes(signer_signature_hash_bytes) else {
+                            debug!("Signer #{}: Received a signature result for a signature over a non-block. Nothing to broadcast.", self.signer_id);
+                            return;
+                        };
+                        let Some(block_info) = self.blocks.remove(&signer_signature_hash) else {
+                            debug!("Signer #{}: Received a signature result for a block we have not seen before. Ignoring...", self.signer_id);
+                            return;
+                        };
+                        block_info.block
+                    });
+                // We don't have enough signers to sign the block. Broadcast a rejection
+                let block_rejection = BlockRejection::new(
+                    block.header.signer_signature_hash(),
+                    RejectCode::InsufficientSigners(malicious_signers.clone()),
+                );
+                debug!(
+                    "Signer #{}: Insufficient signers for block; send rejection {block_rejection:?}",
+                    self.signer_id
+                );
+                // Submit signature result to miners to observe
+                if let Err(e) = self
+                    .stackerdb
+                    .send_message_with_retry(block_rejection.into())
+                {
+                    warn!(
+                        "Signer #{}: Failed to send block submission to stacker-db: {e:?}",
+                        self.signer_id
+                    );
+                }
+            }
+            SignError::Aggregator(e) => {
+                warn!(
+                    "Signer #{}: Received an aggregator error: {e:?}",
+                    self.signer_id
+                );
+            }
+        }
+        // TODO: should reattempt to sign the block here or should we just broadcast a rejection or do nothing and wait for the signers to propose a new block?
```

Comment:
I think ultimately we want the miners to drive block re-submission

Reply:
KK I will broadcast a rejection.